requests_mv_integrations/request_mv_integration.py

```diff
@@ -0,0 +1,1774 @@
+                response_content_encoding = \
+                    safe_str(response.headers['Content-Encoding'])
+                response_extra.update({
+                    'Content-Encoding': response_content_encoding
+                })
+
+            if 'Transfer-Encoding' in response.headers:
+                response_transfer_encoding = \
+                    safe_str(response.headers['Transfer-Encoding'])
+                response_extra.update({
+                    'Transfer-Encoding': response_transfer_encoding
+                })
+
+        if not is_http_status_successful(
+            http_status_code=response.status_code
+        ):
+            self.logger.error(
+                "Validate Response: Failed",
+                extra=response_extra
+            )
+
+            raise TuneRequestModuleError(
+                error_message="Validate Request: Failed",
+                error_request_curl=self.built_request_curl,
+                exit_code=TuneIntegrationExitCode.MOD_ERR_SOFTWARE
+            )
+        else:
+            self.logger.debug(
+                "Validate Response: Success",
+                extra=response_extra
+            )
+
+    def validate_json_response(
+        self,
+        response,
+        request_label=None,
+        response_content_type_expected='application/json',
+        raise_ex_if_not_json_response=True
+    ):
+        """Validate JSON response.
+
+        Args:
+            response:
+            request_label:
+            response_content_type_expected:
+            raise_ex_if_not_json_response:
+
+        Returns:
+
+        """
+        self.validate_response(
+            response,
+            request_label
+        )
+
+        json_response = None
+        response_extra = {}
+        if request_label:
+            response_extra.update({
+                'request_label': request_label
+            })
+
+        response_extra.update({
+            'Content-Type (Expected)': response_content_type_expected
+        })
+
+        if hasattr(response, 'headers'):
+            response_content_type = response.headers.get('Content-Type', None)
+
+        if response_content_type is not None:
+            is_valid_response_content_type = \
+                response_content_type == response_content_type_expected or \
+                response_content_type.startswith(response_content_type_expected)
+
+            if is_valid_response_content_type:
+                json_response = self.requests_response_json(
+                    response=response,
+                    request_label=request_label
+                )
+            elif response_content_type.startswith('text/html'):
+                try:
+                    response_content_html_lines = \
+                        requests_response_text_html(
+                            response=response
+                        )
+                except Exception as ex:
+                    raise TuneRequestModuleError(
```

Comment:
merge this exception with the below one, if the error messages are that important set them with the `if`s and raise at the end